pygt3x/reader.py

```diff
@@ -116,6 +118,16 @@ def get_data(self, num_rows=None):
                 # Idle sleep mode is encoded as an event with payload 8 when entering
                 # and 09 when leaving.
                 if type == Types.Event and evt.payload == b"\x08":
+                    try:
+                        last_second = acceleration[-1][0, 0]
+                    except IndexError:
+                        pass
+                    else:
+                        dt = evt.header.timestamp - last_second
+                        if dt >= 2:
+                            ts = pd.to_datetime(evt.header.timestamp, unit="s")
+                            logging.debug(f"Missed {dt}s before {ts}")
+
```

Comment:
Two checks added:
- each event is required to have sample frequency number of records (`_validate_payload` method); otherwise a ValueError is thrown
- at the end of `get_data()` assertions check if each second appears only sample frequency times in the whole dataset (the previous check doesn't guarantee that this one should pass, because there may be duplicated seconds each with sample frequency records)

Two fixes added:
- fill the gaps right before idle sleep mode is on
- in case the current event timestamp occurs before an already filled-in idle sleep mode period ends, then update the accelerometer with the current event data

Tested on 5 ALS participants.

TODO: 
- Fix 1 failling test `test_read_v1` from `test_formats.py` (fails because 30 Hz data happens to result in 1 record per second). Investigating...
- Add unit tests for the above-mentioned edge cases. I'll have a look if it's possible to simulate these edge cases without the need to provide heavy gt3x files.